src/main/java/org/broadinstitute/consent/http/db/mapper/DacReducer.java

```diff
@@ -0,0 +1,83 @@
+package org.broadinstitute.consent.http.db.mapper;
+
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.util.Map;
+import java.util.Objects;
+import org.broadinstitute.consent.http.models.Dac;
+import org.broadinstitute.consent.http.models.DataAccessAgreement;
+import org.broadinstitute.consent.http.models.Dataset;
+import org.broadinstitute.consent.http.models.FileStorageObject;
+import org.broadinstitute.consent.http.util.ConsentLogger;
+import org.jdbi.v3.core.mapper.MappingException;
+import org.jdbi.v3.core.result.LinkedHashMapRowReducer;
+import org.jdbi.v3.core.result.RowView;
+
+public class DacReducer implements LinkedHashMapRowReducer<Integer, Dac>,
+    ConsentLogger {
+
+  private final DataUseParser dataUseParser = new DataUseParser();
+
+  @Override
+  public void accumulate(Map<Integer, Dac> container, RowView rowView) {
+
+    try {
+
+      Dac dac =
+          container.computeIfAbsent(
+              rowView.getColumn("dac_id", Integer.class), id -> rowView.getRow(Dac.class));
+
+      if (Objects.nonNull(rowView.getColumn("daa_daa_id", Integer.class))) {
```

Comment:
Since the logic is a little different now, I think it's worth adding some unit tests to cover the new lines.

Reply:
Would these unit tests be in the DacDAOTest class?

Reply:
Yes, I think it would make sense to add in that class.